### PR TITLE
Feature/data collection storage

### PR DESCRIPTION
**Pull request by zbrydon**
# Data Collection and Storage

## Overview 
- Added game playtime and rating data collection and storage
- Utilised SplashKits inbuilt SQLite database 

**Riley** 
- Created the rating class which collects the users rating
- Implemented complex query to aggregate stats
- Show rating on the game overlay
- Show playtime on the game overlay
- Created generic function to query the database
- Styled the rating form
- FIX: Stop displaying tips when game has started

**Zac**
- Created GameData class to store individual and aggregate game statistics 
- Created Database class to abstract database functions from the GameData class
- Created Table class to store each of a databases tables

## To Do
- Refactor some of the database functions slightly potentially

### CODE REVIEW

**Review by studioant — COMMENTED** [review]
This is a great contribution to the arcade machine, opening many avenues for expansion now that data can be stored. Excellent work!

File: Rating.h [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        {
+            _rating--;
+            if (_rating < 0)
+                _rating = _maxRating;
+            updateGrid();
+        }
+        else if (key_typed(RIGHT_KEY))
+        {
+            // rating is between 1 and 5
+            _rating = ++_rating % (_maxRating + 1);
```
[the remark this answers is not in the document]
Comment (Delcari) [reply Rating.h]: Resolved in [3944e4f](https://github.com/thoth-tech/arcade-machine/pull/49/commits/3944e4f65d0c78c1b76afddf5d836808b21473dc)

File: Menu.h [hunk 2]
```diff
@@ -1,555 +1,591 @@
...
+        Table *gameDataTable = new Table("gameData", {{"gameName", "TEXT"}, {"startTime", "TEXT"}, {"endTime", "TEXT"}, {"rating", "TEXT"}, {"highScore", "TEXT"}});
+        this->_db->createTable(gameDataTable);
+        this->_rating = Rating();
+    }
+
+    Menu(std::vector<ConfigData> configs)
+    {
+        this->_games = configs;
+        this->_db = new Database();
+        Table *gameDataTable = new Table("gameData", {{"gameName", "TEXT"}, {"startTime", "TEXT"}, {"endTime", "TEXT"}, {"rating", "TEXT"}, {"highScore", "TEXT"}});
```
[the remark this answers is not in the document]
Comment (Delcari) [reply Menu.h:105]: Discussed

**Review by lachfoy — APPROVED** [review]
Looks great, have tested on Windows and all features are behaving as expected. 👍 !

**Pull request merged**